src/components/User/OpinionsAdmin.js

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react';
+import { IoClose } from 'react-icons/io5';
+import { GiCheckMark } from 'react-icons/gi';
+import { BiHide, BiShow } from 'react-icons/bi';
+import { collection, deleteDoc, doc, onSnapshot, query, updateDoc } from 'firebase/firestore';
+import { db } from '../configuration/firebase';
+
+function OpinionsAdmin() {
+  const [allOpinions, setAllOpinions] = useState([]);
+
+  useEffect(() => {
+    const q = query(collection(db, 'opinions'));
+    const unsubscribe = onSnapshot(q, (querySnapshot) => {
+      const opinions = [];
+      querySnapshot.forEach((document) => {
+        opinions.push({ doc_id: document.id, ...document.data() });
+      });
+      setAllOpinions(opinions);
+    });
+    return () => unsubscribe();
```

Comment:
Tu chyba można bezpośrednio zwrócić tą funkcję unsubscribe, zamiast tworzyć stałą wcześniej..

Reply:
W ten sposob wydaje mi sie bardziej czytelnie i nie wydaje mi się ze to jakis duzy błąd